streams/piping/abort.any.js

```diff
@@ -115,61 +118,70 @@ promise_test(t => {
     });
 }, 'preventCancel and preventAbort should prevent canceling the readable and aborting the readable');
 
-promise_test(t => {
-  const rs = new ReadableStream({
-    start(controller) {
-      controller.enqueue('a');
-      controller.enqueue('b');
-      controller.close();
-    }
-  });
-  const abortController = new AbortController();
-  const signal = abortController.signal;
-  const ws = recordingWritableStream({
-    write() {
-      abortController.abort();
-    }
-  });
-  return promise_rejects_dom(t, 'AbortError', rs.pipeTo(ws, { signal }), 'pipeTo should reject')
-      .then(() => {
-        assert_equals(ws.events.length, 4, 'only chunk "a" should have been written');
-        assert_array_equals(ws.events.slice(0, 3), ['write', 'a', 'abort'], 'events should match');
-        assert_equals(ws.events[3].name, 'AbortError', 'abort reason should be an AbortError');
-      });
-}, 'abort should prevent further reads');
+for (const reason of [null, undefined, error1]) {
+  promise_test(t => {
+    let error;
+    const rs = new ReadableStream({
+      start(controller) {
+        controller.enqueue('a');
+        controller.enqueue('b');
+        controller.close();
+      }
+    });
+    const abortController = new AbortController();
+    const signal = abortController.signal;
+    const ws = recordingWritableStream({
+      write() {
+        abortController.abort(reason);
+      }
+    });
+    return rs.pipeTo(ws, { signal })
+        .catch(e => { error = e; })
```

Comment:
[We used to check](https://github.com/web-platform-tests/wpt/blob/e1e713c842e54ea0a9410ddc988b63d0e1d31973/streams/piping/abort.any.js#L133) that aborting the signal causes the stream to error with an `AbortError` DOMException. In the updated test, we don't assert anything about the *contents* of this `error`.

Can we still assert *something* about `error`? For example, if `reason` is `error1`, then we expect `error` to also be `error1`. And if `reason` is `null` or `undefined`, we expect `error` to be a newly constructed `AbortError`, correct?